src/middlewares/auth.middleware.ts

```diff
@@ -0,0 +1,56 @@
+import {Request, Response, NextFunction} from "express";
+import {JWT} from "@/config";
+import jwt from "jsonwebtoken";
+
+declare module "express-serve-static-core" {
+    interface Request {
+        user?: {id: number, role: string, email?: string}
+    }
+}
+
+export async function authMiddleware(
+    req: Request, 
+    res:Response, 
+    next: NextFunction){
+
+        try{
+            const authHeader = req.headers["authorization"];
+
+            if (!authHeader || !authHeader.startsWith("Bearer ")){
+                return res.status(401).json({message: "Authorization token required"});
+            }
+
+            const token = authHeader.split( " " ) [1];
+
+            const decodedToken = jwt.verify(token, JWT.secret) as {
```

Comment:
you created an interface during token creation. why are you casting into a different object during verification. and i dont remember you encoding email during creation..so how did email come in here

and even at that, email is required during account creation, if you decide to put email as part of the token, why are you making it possibly undefined with "?"